lib/src/js/value/color.dart

```diff
@@ -72,7 +357,54 @@ final JSClass colorClass = () {
 ///
 /// This ensures that an explicitly null alpha will be treated as a missing
 /// component.
-num? _handleUndefinedAlpha(num? alpha) => isUndefined(alpha) ? 1 : alpha;
+double? _handleUndefinedAlpha(double? alpha) => isUndefined(alpha) ? 1 : alpha;
+
+/// This procedure takes a `channel` name, an object `changes` and a SassColor
+/// `initial` and returns the result of applying the change for `channel` to
+/// `initial`.
+double? _changeComponentValue(
+        SassColor initial, String channel, _ConstructionOptions changes) =>
+    hasProperty(changes, channel)
```

Comment:
We want to allow authors to explicitly pass `undefined` to say "use the previous channel value", which means checking whether the property exists isn't sufficient to tell whether you should use the original value. You can use the `isUndefined()` function from `lib/src/js/utils.dart` instead.

This is probably worth adding a test for.

Reply:
I updated this in edc9b2cfed9d656ceb1522df4f5896e43de34a9d, and added tests to sass-spec in https://github.com/sass/sass-spec/pull/1949/commits/863047107776590ddef78bc225df956be3da7aad, and updated the spec to reflect this in https://github.com/sass/sass/pull/3723/commits/030aa3eba37de8324ed2b80bb0807fbc0f638201.

> This is probably worth adding a test for.

This has higher-level test coverage in sass-spec, but are you recommending unit tests within the `dart-sass` repo?

Reply:
No, I meant in sass-spec